hnn_core/network_models.py

```diff
@@ -31,6 +31,8 @@ def jones_2009_model(params=None, add_drives_from_params=False,
         are added suitably. Will be deprecated in a future release.
     mesh_shape : tuple of int (default: (10, 10))
         Defines the (n_x, n_y) shape of the grid of pyramidal cells.
+    allow_autapses : Set to True by default (as in the original paper).
+        Allows cells to connect to themselves. Set to False for L5_Basket.
```

Comment:
```suggestion
    allow_autapses : bool
        If True, self-connections are enabled in all cell populations.
```

just rephrased it so it follows numpy docstring conventions and stays consistent with remaining repo.


is it only L5 Basket cells which allow autapses? So other cell populations remain unaffected by this change?

Reply:
L5 basket cells and L5 pyramidal cells don't allow autapses, the others do.

@katduecker I'm realizing that that parameter is only invoked when there is a risk of autapses forming when you call `net.add_connection`. If you were to connect L5Pyr->any other cell, that parameter has not impact because it's only job is to prevent a gid connecting to itself.

I don't think it changes the logic of your PR, but I'm realizing that for most of those `net.add_connection` calls it could be set to true or false and not make a difference.